tracer/src/Datadog.Trace/Logging/DirectSubmission/Formatting/LogFormatter.cs

```diff
@@ -32,38 +32,43 @@ internal class LogFormatter
         private readonly string? _env;
         private readonly string? _version;
         private readonly IGitMetadataTagsProvider _gitMetadataTagsProvider;
-        private string? _tags;
         private bool _gitMetadataAdded;
 
         private string? _ciVisibilityDdTags;
 
         public LogFormatter(
-            ImmutableDirectLogSubmissionSettings settings,
+            ImmutableTracerSettings settings,
+            ImmutableDirectLogSubmissionSettings directLogSettings,
             ImmutableAzureAppServiceSettings? aasSettings,
             string serviceName,
             string env,
             string version,
             IGitMetadataTagsProvider gitMetadataTagsProvider)
         {
             _gitMetadataTagsProvider = gitMetadataTagsProvider;
-            _source = string.IsNullOrEmpty(settings.Source) ? null : settings.Source;
+            _source = string.IsNullOrEmpty(directLogSettings.Source) ? null : directLogSettings.Source;
             _service = string.IsNullOrEmpty(serviceName) ? null : serviceName;
-            _host = string.IsNullOrEmpty(settings.Host) ? null : settings.Host;
-            _tags = EnrichTagsWithAasMetadata(settings.GlobalTags, aasSettings);
+            _host = string.IsNullOrEmpty(directLogSettings.Host) ? null : directLogSettings.Host;
+
+            var globalTags = directLogSettings.GlobalTags is { Count: > 0 } ? directLogSettings.GlobalTags : settings.GlobalTagsInternal;
+
+            Tags = EnrichTagsWithAasMetadata(ImmutableDirectLogSubmissionSettings.StringifyGlobalTags(globalTags), aasSettings);
```

Comment:
nit: I don't think there's any reason for `StringifyGlobalTags` to live on `ImmutableDirectLogSubmissionSettings` now? I would move it to `LogFormatter` instead

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/4902/commits/69541684633cdfbab014eb59c5240334490343e0